version: "3.8"

services:
  flask_website:
    build:
      context: .
      dockerfile: Flask.Dockerfile
    container_name: flask-container
    expose:
      - "5005"
    ports:
      - "5005:5005"
    networks:
      - app-network

  nginx:
    build:
      context: .
      dockerfile: Nginx.Dockerfile
    container_name: nginx-container
    ports:
      - "80:80"
      - "443:443"
    depends_on:
      - flask_website
    networks:
      - app-network
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
      - certbot_certs:/etc/letsencrypt
      - certbot_var:/var/lib/letsencrypt

  certbot:
    image: certbot/certbot
    container_name: certbot-container
    depends_on:
      - nginx
    networks:
      - app-network
    volumes:
      - certbot_certs:/etc/letsencrypt
      - certbot_var:/var/lib/letsencrypt
      - ./certbot-www:/var/www/certbot
    entrypoint: >
      sh -c "
      certbot certonly --webroot
      --webroot-path=/var/www/certbot
      --email jbur645@wgu.edu
      --agree-tos
      --no-eff-email
      -d www.django-blog-jb.com &&
      nginx -s reload" 

networks:
  app-network:
    driver: bridge

volumes:
  certbot_certs:
  certbot_var: